src/pathwayvis/components/reactionpanel/objectives/objectives.component.ts

```diff
@@ -0,0 +1,80 @@
+import * as angular from "angular";
+
+import { MapOptionService } from "../../../services/mapoption.service";
+import { ActionsService } from "../../../services/actions/actions.service";
+
+const DecafBiggProxy = 'https://api-staging.dd-decaf.eu/bigg/';
+
+class ObjectivesController {
+  public modelReactions = [];
+  public objectiveReaction: string[] = [];
+  private action: string = '';
+  private _mapOptions: MapOptionService;
+  private _actions: ActionsService;
+
+  constructor(mapOptions: MapOptionService,
+    actions: ActionsService) {
+    this._mapOptions = mapOptions;
+    this._actions = actions;
+
+    mapOptions.reactionsObservable.subscribe((reactions) => {
+      this.modelReactions = reactions;
+    });
+
+    mapOptions.objectiveReactionObservable.subscribe((reaction) => {
+      this.objectiveReaction = [];
```

Comment:
Why do you set it to empty list? You could just use a simple element instead of an array, and set it to `null` if it is removed. Or are you planning to support multiple objectives?